roblox/jobs.py

```diff
@@ -144,3 +146,123 @@ def __init__(self, client: Client, data: dict):
             ) for instance_data in data["Collection"]
         ]
         self.total_collection_size: int = data["TotalCollectionSize"]
+
+
+class ServerType(Enum):
+    """
+    Represents the type of server.
+    """
+    
+    Public = 0
+    Friend = 1
+
+
+class ServerPlayer(BaseUser):
+    """
+    Represents a player in a server.
+    
+    Attributes:
+        id: The player's user id.
+        name: The player's username.
+        displayName: The player's display name.
+        player_token: The player's token.
+    """
+
+    def __init__(self, client: Client, data: dict):
+        """
+        Arguments:
+            client: The Client this object belongs to.
+            data: A GameServerPlayerResponse object.
+        """
+
+        super().__init__(client=client, user_id=data["id"])
+
+        self.player_token: str = data["playerToken"]
+        self.name: str = data["name"]
+        self.display_name: str = data["displayName"]
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} id={self.id} name={self.name} display_name={self.display_name} player_token={self.player_token}>"
+
+
+class Server:
+    """
+    Represents a public server.
+
+    Attributes:
+        id: The server's job id.
+        max_players: The maximum number of players that can be in the server at once.
+        playing: The amount of players in the server.
+        player_tokens: A list of thumbnail tokens for all the players in the server.
+        players: A list of ServerPlayer objects representing the players in the server. Only friends of the authenticated user will show up here.
+        fps: The server's fps.
+        ping: The server's ping.
+    """
+
+    def __init__(self, client: Client, data: dict):
+        """
+        Arguments:
+            client: The Client this object belongs to.
+            data: A GameServerResponse object.
+        """
+
+        self._client: Client = client
+
+        self.id: Optional[str] = data.get("id")
+        self.max_players: int = data["maxPlayers"]
+        self.playing: int = data.get("playing", 0)
+        self.player_tokens: List[str] = data["playerTokens"]
+        self.players: List[ServerPlayer] = [
+            ServerPlayer(client=self._client, data=player_data) 
+            for player_data in data["players"]
+        ]
+
+        self.fps: float = data.get("fps")
+        self.ping: Optional[int] = data.get("ping")
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} id={self.id} playing={self.playing}>"
```

Comment:
This and `__eq__` and `__ne__` should be implemented by a base class like BaseJob or at the very least BaseItem